File: leetcode/0509. Fibonacci Number/Solution.swift
// https://leetcode.com/problems/fibonacci-number/

class Solution {
    func fib(_ N: Int) -> Int {
        var a = 0
        var b = 1
        for _ in 0..<N {
            let t = a + b
            a = b
            b = t
        }
            
        return a
    }
}

File: leetcode/0509. Fibonacci Number/Solution.py
# https://leetcode.com/problems/fibonacci-number/

class Solution:
    def fib(self, N: int) -> int:
        a = 0
        b = 1
        for _ in range(N):
            t = a + b
            a = b
            b = t
            
        return a

s = Solution()
assert s.fib(2) == 1
assert s.fib(3) == 2
assert s.fib(4) == 3